src/content/docs/cheerpj3/12-reference/00-cheerpjInit.md

```diff
@@ -226,6 +226,68 @@ Example usage:
 cheerpjInit({ javaProperties: ["prop1=value1", "prop2=value2"] });
 ```
 
+### `tailscaleControlUrl`
+
+```ts
+tailscaleControlUrl?: string;
+```
+
+This option expects a string URL of the Tailscale control plane. The control plane verifies the user's identity and validates various keys among the connected peers in the network. This option is used in the scenario of self-hosting, if omitted it will point to the Tailscale control plane.
+
+Example usage:
+
+```js
+cheerpjInit({ tailscaleControlUrl: "https://my.url.com/" });
+```
+
+### `tailscaleDnsUrl`
+
+```ts
+ tailscaleDnsUrl?: string;
+```
+
+Expects a string with the IPv4 or IPv6 address to use for DNS queries. If omitted, the default IP address is "8.8.8.8". For more information visit [DNS in Tailscale](https://tailscale.com/kb/1054/dns/).
+
+Example usage:
+
+```js
+cheerpjInit({ tailscaleDnsUrl: "1.1.1.1" });
+```
+
+### `tailscaleAuthKey`
+
+```ts
+tailscaleAuthKey?: string;
+```
+
+This option expects a string that contains a Tailscale auth key. Using auth keys allows to register new nodes that are pre-authenticated. You can create an auth key [here](https://login.tailscale.com/admin/settings/keys). This option is mutually exclusive with [`tailscaleLoginUrlCb`](#tailscaleloginurlcb)
+
+For more information about auth keys visit the [Tailscale auth keys documentation](https://tailscale.com/kb/1085/auth-keys/).
+
+Example of usage:
+
+```js
+cheerpjInit({ tailscaleAuthKey: "AuthKeyStringGoesHere" });
+```
+
+### `tailscaleLoginUrlCb`
+
+```ts
+tailscaleLoginUrlCb?: (r: string) => void;
+```
+
+This option is used to login into a Tailscale network and it is mutually exclusive with [`tailscaleAuthKey`](#tailscaleauthkey). It expects the base URL of a control server that will continue and finish the login process. This callback is executed when it is time to prompt the user to login to Tailscale via the UI.
+
```

Comment:
Say it's mutually exclusive with tailscaleAuthKey

Reply:
I think I did o.o

Reply:
oops, my bad!